src/plugins/intel_npu/src/backend/src/zero_tensor.cpp

```diff
@@ -0,0 +1,144 @@
+        OPENVINO_THROW("Tensor data with element type ",
+                       get_element_type(),
+                       ", is not representable as pointer to ",
+                       element_type);
+    }
+    return _ptr;
+}
+
+const ov::element::Type& ZeroTensor::get_element_type() const {
+    return _element_type;
+}
+
+const ov::Shape& ZeroTensor::get_shape() const {
+    return _shape;
+}
+
+void ZeroTensor::update_strides() const {
+    if (_element_type.bitwidth() < 8)
+        return;
+
+    auto& shape = get_shape();
+    if (_strides.empty() && !shape.empty()) {
+        _strides.resize(shape.size());
+        _strides.back() = shape.back() == 0 ? 0 : _element_type.size();
+        std::transform(shape.crbegin(),
+                       shape.crend() - 1,
+                       _strides.rbegin(),
+                       _strides.rbegin() + 1,
+                       std::multiplies<size_t>());
+    }
+}
+
+const ov::Strides& ZeroTensor::get_strides() const {
+    OPENVINO_ASSERT(_element_type.bitwidth() >= 8,
+                    "Could not get strides for types with bitwidths less then 8 bit. Tensor type: ",
+                    _element_type);
+    std::call_once(_strides_once, &ZeroTensor::update_strides, this);
+    return _strides;
+}
+
+void ZeroTensor::initialize_elements(void* data, const ov::element::Type& element_type, const ov::Shape& shape) {
+    if (element_type == ov::element::Type_t::string) {
+        auto num_elements = shape_size(shape);
+        auto string_ptr = static_cast<std::string*>(data);
+        std::uninitialized_fill_n(string_ptr, num_elements, std::string());
+    }
+}
+
+size_t ZeroTensor::get_capacity() const {
+    return shape_size(_capacity);
+}
+
+size_t ZeroTensor::get_bytes_capacity() const {
+    return ov::element::get_memory_size(get_element_type(), get_capacity());
+}
+
+void ZeroTensor::destroy_elements(size_t begin_ind, size_t end_ind) {
+    // it removes elements from tail
+    if (get_element_type() == ov::element::Type_t::string) {
+        auto strings = static_cast<std::string*>(_ptr);
+        for (size_t ind = begin_ind; ind < end_ind; ++ind) {
+            using std::string;
+            strings[ind].~string();
+        }
+    }
+}
+
+void ZeroTensor::destroy_memory() {
+    destroy_elements(0, get_capacity());
+    _allocator.deallocate(_ptr, get_bytes_capacity());
+    _ptr = nullptr;
+}
+
+void ZeroTensor::set_shape(ov::Shape new_shape) {
+    if (_shape == new_shape)
+        return;
+
+    _shape = std::move(new_shape);
+
+    if (get_size() > get_capacity()) {
+#ifdef __linux__
+        OPENVINO_THROW("Re-shaping the tensor with a larger shape is not available.");
+#endif
+
+        if (!_init_structs->getMutableCommandListVersion()) {
+            OPENVINO_THROW("Re-shaping the tensor with a larger shape is not available using this driver version. "
+                           "Please update the driver.");
```

Comment:
Wondering if there's a minimum version we can recommend without ambiguity. Or if there's no point in doing so, then maybe we could specify updating to the latest one.